File: Ai2c Apps/Amap-backend-dev/utils/tests/unit.py
from units.models import Unit


def create_testing_unit(
    uic: str = "TEST000AA",
    short_name: str = "1-100 TEST",
    display_name: str = "Unit Display Name",
    nick_name: str = "Unit Nick Name",
    echelon: Unit.Echelon = Unit.Echelon.ARMY,
    # logo:None = None,
    compo: Unit.Component = Unit.Component.ACTIVE,
    state: str = "UNK",
    parent_unit: Unit | None = None,
    parent_uics: list[str | None] = [],
    level: int = 0,
    child_uics: list[str | None] = [],
    subordinate_uics: list[str | None] = [],
    as_of_logical_time: int = 0,
) -> Unit:
    """
    This will create a Unit object from units.models.Unit

        @param uic: (str) The primary key for the Unit
        @param short_name: (str)
        @param display_name: (str)
        @param nick_name: (str)
        @param echelon: (Unit.Echelon)
        @param #logo: ()
        @param compo: (Unit.Component)
        @param state: (str)
        @param parent_unit: (Unit)
        @param parent_uics: (list[str | None])
        @param child_uics: (list[str | None])
        @param subordinate_uics: (list[str | None])
        @param as_of_logical_time: int

        @returns Unit
    """
    new_unit = Unit.objects.create(
        uic=uic,
        short_name=short_name,
        display_name=display_name,
        nick_name=nick_name,
        echelon=echelon,
        # logo = logo,
        compo=compo,
        state=state,
        parent_unit=parent_unit,
        parent_uics=parent_uics,
        level=level,
        child_uics=child_uics,
        subordinate_uics=subordinate_uics,
        as_of_logical_time=as_of_logical_time,
    )

    new_unit.set_all_unit_lists()
    if parent_unit:
        parent_unit.set_all_unit_lists()

    return new_unit

File: Ai2c Apps/frontend-react-griff/frontend-react-dev/src/pmx-mui/models/Unit.ts
import { Echelon } from './Echelon';

/**
 * @typedef UnitDto
 * @prop
 */
export interface UnitDto {
  uic: string;
  short_name: string;
  display_name: string;
  nick_name: string | null;
  echelon: Echelon | string;
  compo: string;
  state?: string | null;
  parent_uic: string | null;
  level: number;
}

/**
 * @typedef Unit
 * @prop
 */
export interface Unit {
  uic: string;
  shortName: string;
  displayName: string;
  nickname: string | null;
  echelon: Echelon | string;
  component: string;
  state?: string | null;
  parentUic: string | undefined;
  level: number;
}

/**
 * mapUnitDtoToUnit
 *
 * Map Unit Api DTO response to Unit Model
 *
 * @param { UnitDto } dto API response
 * @return { Unit } Local model
 */
export const mapUnitDtoToUnit = (dto: UnitDto): Unit => {
  return {
    uic: dto.uic,
    parentUic: dto.parent_uic ?? '',
    displayName: dto.display_name,
    shortName: dto.short_name,
    nickname: dto.nick_name,
    echelon: dto.echelon,
    component: dto.compo,
    state: dto.state,
    level: dto.level,
  };
};
